packages/components/molecules/f-rating/src/components/Rating.vue

```diff
@@ -2,25 +2,38 @@
     <div
         :class="$style['c-rating']"
         data-test-id="rating-component">
-        <ul
-            :class="$style['c-rating-starWrapper']">
-            <li
-                v-for="star in maxStarRating"
-                :key="star"
-                :class="$style['c-rating-star']">
-                <star-filled-icon
-                    v-if="isRatingStarFilled(star)"
-                    :class="$style['c-rating-star-filled']" />
+        <div :class="$style['c-rating-starWrapper']">
+            <div :class="$style['c-rating-container']">
                 <star-icon
-                    v-else
-                    :class="$style['c-rating-star-empty']" />
-            </li>
-        </ul>
-        <span
-            data-test-id="c-rating-description"
-            class="is-visuallyHidden">
-            {{ getRatingDescription }}
-        </span>
+                    v-for="star in maxStarRating"
+                    :key="star"
+                    :class="[
+                        $style['c-rating-star-empty'],
+                        $style[`c-rating-star--${starRatingSize}`]
+                    ]" />
+            </div>
+
+            <div
+                :class="[
+                    $style['c-rating-mask'],
+                    $style['c-rating-container']
+                ]"
+                :style="getRatingStarPercentage">
```

Comment:
follow up from my comment below. This could be something like:

```
:style="`--starRatingPercentage: ${getRatingStarPercentage}`"
```

Just guessing the syntax there but I think it's close 🤣

Reply:
Didn't know you could do that. Ill give that a try 👍